[CI][dashboard] Add a workflow to collect A10g perf

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131816

Summary: This is an experimental work. Depending on the performance stableness and benchmark coverage on A10g, we may consider to use A10g for manually-triggered per-PR performance comparison instead of exausting expensive A100 instances.

desertfire committed:
[CI][dashboard] Add a workflow to collect A10g perf

Summary: This is an experimental work. Depending on the performance stableness and benchmark coverage on A10g, we may consider to use A10g for manually-triggered per-PR performance comparison instead of exausting expensive A100 instances.

[ghstack-poisoned]

huydhn reviewed (approved):
Let me draft the corresponding dashboard change

huydhn commented on .github/workflows/inductor-perf-test-nightly-a10g.yml:
```diff
@@ -0,0 +1,125 @@
+      cuda-arch-list: '8.0'
+      test-matrix: |
+        { include: [
+          { config: "inductor_huggingface_perf_A10g", shard: 1, num_shards: 3, runner: "linux.g5.4xlarge.nvidia.gpu" },
```
I think we need to call this `cuda_a10g` instead, same as what you have in https://github.com/pytorch/pytorch/pull/131729.  Otherwise, the device name will just be `cuda` and there is no easy way to differentiate between `cuda_a100` and `cuda_a10g` records.

huydhn replied:
The dashboard change https://github.com/pytorch/test-infra/pull/5505

desertfire committed:
Update on "[CI][dashboard] Add a workflow to collect A10g perf"


Summary: This is an experimental work. Depending on the performance stableness and benchmark coverage on A10g, we may consider to use A10g for manually-triggered per-PR performance comparison instead of exausting expensive A100 instances.

[ghstack-poisoned]

desertfire committed:
Update on "[CI][dashboard] Add a workflow to collect A10g perf"


Summary: This is an experimental work. Depending on the performance stableness and benchmark coverage on A10g, we may consider to use A10g for manually-triggered per-PR performance comparison instead of exausting expensive A100 instances.

[ghstack-poisoned]

desertfire committed:
Update on "[CI][dashboard] Add a workflow to collect A10g perf"


Summary: This is an experimental work. Depending on the performance stableness and benchmark coverage on A10g, we may consider to use A10g for manually-triggered per-PR performance comparison instead of exausting expensive A100 instances.

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -f "only affects this new CI workflow"
